scheduler/src/cook/mesos/share.clj

```diff
@@ -146,13 +175,11 @@
                                  :where
                                  [?q :share/user ?user]]
                                db)
-          user->share-cache (->> all-share-users
-                                 (map (fn [user]
-                                        [user (get-share db user)]))
-                                 ;; In case default-user doesn't have an explicit share
-                                 (cons [default-user (get-share db default-user)])
-                                 (into {}))]
+          default-user-share (get-share db default-user)
+          user->share-cache (-> (get-shares db all-share-users)
+                                ;; In case default-user doesn't have an explicit share
+                                (assoc default-user default-user-share))]
       (fn user->share
```

Comment:
I measured locally, this change doesn't have a noticeable impact `sort-jobs-by-dru` (and hence `rank-jobs`). Runs for the new and old variants of `sort-jobs-by-dru` were running in about 4 seconds.